test/integration/elasticquota_controller_test.go

```diff
@@ -0,0 +1,355 @@
+		groupList, _, err := cs.ServerGroupsAndResources()
+		if err != nil {
+			return false, nil
+		}
+		for _, group := range groupList {
+			if group.Name == scheduling.GroupName {
+				return true, nil
+			}
+		}
+		t.Log("waiting for crd api ready")
+		return false, nil
+	}); err != nil {
+		t.Fatalf("Waiting for crd read time out: %v", err)
+	}
+
+	// start controller
+	stopCh := server.SetupSignalHandler()
+	go eqCtrl.Run(1, ctx.Done())
+	schedInformerFactory.Start(stopCh)
+	coreInformerFactory.Start(stopCh)
+
+	testCtx.ClientSet = cs
+	testCtx = util.InitTestSchedulerWithOptions(
+		t,
+		testCtx,
+		true,
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+	t.Log("init scheduler success")
+	defer testutil.CleanupTest(t, testCtx)
+
+	// Create a Node.
+	nodeName := "fake-node"
+	node := st.MakeNode().Name(nodeName).Label("node", nodeName).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(3000, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(3000, resource.DecimalSI),
+	}
+	node, err = cs.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName, err)
+	}
+
+	for _, ns := range []string{"ns1", "ns2"} {
+		_, err := cs.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: ns}}, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to integration test ns: %v", err)
+		}
+		autoCreate := false
+		t.Logf("namespaces %+v", ns)
+		_, err = cs.CoreV1().ServiceAccounts(ns).Create(ctx, &v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{Name: "default", Namespace: ns}, AutomountServiceAccountToken: &autoCreate}, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to create ns default: %v", err)
+		}
+	}
+
+	for _, tt := range []struct {
+		name          string
+		elasticQuotas []*v1alpha1.ElasticQuota
+		nowPods       []*v1.Pod
+		nowEQUsed     []*v1alpha1.ElasticQuota
+		nextPods      []*v1.Pod
+		nextEQUsed    []*v1alpha1.ElasticQuota
+	}{
+		{
+			name: "The status of the pod changes from pending to running",
+			elasticQuotas: []*v1alpha1.ElasticQuota{
+				MakeEQ("ns1", "t1-eq1").
+					Min(MakeResourceList().CPU(100).Mem(1000).Obj()).
+					Max(MakeResourceList().CPU(100).Mem(1000).Obj()).Obj(),
+				MakeEQ("ns2", "t1-eq2").
+					Min(MakeResourceList().CPU(100).Mem(1000).Obj()).
+					Max(MakeResourceList().CPU(100).Mem(1000).Obj()).Obj(),
+			},
+			nowPods: []*v1.Pod{
+				MakePod("ns1", "t1-p1").Node("none").
```

Comment:
what's the intention of `Node("none")`? skip scheduling and assign it to a non-existing node?

Reply:
No motivation, just wanted to construct the pending state, but also didn't need to specify the node. I've removed it

BTW:
In fact, we can indeed skip the scheduling phase because it is a test for the controller and also the eq controller is only interested in the running pod.